app/scripts/lib/transaction/hooks/delegation-7702-publish.ts

```diff
@@ -165,97 +186,89 @@ export class Delegation7702PublishHook {
     };
   }
 
-  #buildExecutions(
+  async #isBridgeTxGasless7702(
     transactionMeta: TransactionMeta,
-    gasFeeToken: GasFeeToken,
-  ): Execution[] {
-    const { txParams } = transactionMeta;
-    const { data, to, value } = txParams;
-
-    const userExecution: Execution = {
-      target: to as Hex,
-      value: (value as Hex) ?? '0x0',
-      callData: (data as Hex) ?? EMPTY_HEX,
-    };
-
-    const transferExecution: Execution = {
-      target: gasFeeToken.tokenAddress,
-      value: '0x0',
-      callData: this.#buildTokenTransferData(
-        gasFeeToken.recipient,
-        gasFeeToken.amount,
-      ),
-    };
-
-    return [userExecution, transferExecution];
+  ): Promise<boolean> {
+    try {
+      const state = (await this.#messenger.call(
+        'BridgeStatusController:getState',
+      ));
+      const gasIncluded7702 =
+        state?.submissionRequests?.[transactionMeta.chainId]?.[transactionMeta.id]
+          ?.quoteResponse?.quote?.gasIncluded7702;
+      return Boolean(gasIncluded7702);
+    } catch (error) {
+      log(
+        'Failed to determine whether the quoted transaction is intended to be gasless through EIP-7702',
+        error,
+      );
+      return false;
+    }
   }
 
-  async #buildDelegation(
+  #buildUnsignedDelegation(
```

Comment:
Why have we removed the signing from here?

It seemed a like nice abstraction to build it all inside a method.